src/library/EndpointManagement/EndpointManagementClient.cs

```diff
@@ -494,6 +454,84 @@ private async Task EnsureEndpointManagerRunningAsync(CancellationToken cancellat
+            }
+
+            return null;
+        }
+
+        public async Task<(string fileName, string command)> GetEndpointManagerLaunchArguments(string currentUserName, string logFileDirectory, CancellationToken cancellationToken)
+        {
+            var fileName = string.Empty;
+            var command = string.Empty;
+            var executableName = $"{EndpointManager.ProcessName}{(_platform.IsWindows ? ".exe" : string.Empty)}";
+            var launcherPath = _fileSystem.Path.Combine(_fileSystem.Path.GetExecutingAssemblyDirectoryPath(), EndpointManager.DirectoryName, executableName);
+            if (!_fileSystem.FileExists(launcherPath))
+            {
+                throw new InvalidOperationException($"Failed to find '{launcherPath}'.");
+            }
+
+            switch (_platform)
+            {
+                case var platform when platform.IsWindows:
+                    // Since EndpointManager is started as an elevated process and when a DOTNET_ROOT env variable is set,
+                    // on windows it is not possible to pass the DOTNET_ROOT value, so using EndpointManagerLauncher to start EndpointManager.
+                    // When VSCode uses BinariesV2 strategy to download clients, DOTNET_ROOT env var is set.
+                    if (!string.IsNullOrEmpty(this._environmentVariables.DotNetRoot))
+                    {
+                        var epmLauncherPath = _fileSystem.Path.Combine(_fileSystem.Path.GetExecutingAssemblyDirectoryPath(), EndpointManager.LauncherDirectoryName, executableName);
+                        var quotedArguments = $"\"{this._environmentVariables.DotNetRoot.Trim('"')}\" \"{launcherPath}\" \"{currentUserName}\" \"{_socketFilePath}\" \"{logFileDirectory}\" \"{_operationContext.CorrelationId}\"";
+                        // Note: If the UAC prompt is declined, the process will throw a Win32Exception.
+                        Process.Start(this.GetEndpointManagerLaunchInfoWindows($"\"{epmLauncherPath}\"", quotedArguments));
+                    }
+                    else
+                    {
+                        var quotedArguments = $"\"{currentUserName}\" \"{_socketFilePath}\" \"{logFileDirectory}\" \"{_operationContext.CorrelationId}\"";
+                        // Note: If the UAC prompt is declined, the process will throw a Win32Exception.
+                        Process.Start(this.GetEndpointManagerLaunchInfoWindows($"\"{launcherPath}\"", quotedArguments));
+                    }
+
+                    await CheckEndpointManagerAliveAsync(cancellationToken);
+
+                    break;
+
+                case var platform when platform.IsOSX:
+
+                    var quotedLaunchPathAndArguments = $"\\\\\\\"{launcherPath}\\\\\\\" \\\\\\\"{currentUserName}\\\\\\\" \\\\\\\"{_socketFilePath}\\\\\\\" \\\\\\\"{logFileDirectory}\\\\\\\" \\\\\\\"{_operationContext.CorrelationId}\\\\\\\"";

```

Comment:
I know it was already like this, but...given the number of backslashes it might be clearer to use a verbatim string literal, like:
```csharp
var quotedLaunchPathAndArguments = @$"\\\""{launcherPath}\\\"" \\\""{currentUserName}\\\"" \\\""{_socketFilePath}\\\"" \\\""{logFileDirectory}\\\"" \\\""{_operationContext.CorrelationId}\\\""";
```